src/main/java/org/jenkinsci/plugins/verifystatus/VerificationsPublisher.java

```diff
@@ -100,65 +109,61 @@ public String getVerifyStatusRerun() {
     return verifyStatusRerun;
   }
 
+  public Short getVerifyStatusValue() {
+    return verifyStatusValue;
+  }
+  @DataBoundSetter
+  public void setVerifyStatusValue(Short value) {
+    this.verifyStatusValue=value;
+  }
   // publish report after job completes to get correct duration
   @Override
   public boolean needsToRunAfterFinalized() {
     return true;
   }
 
-  private GerritTrigger getGerritTrigger(AbstractBuild<?, ?> build) {
-    AbstractProject<?,?> project = build.getProject();
-    Iterator<?> it = project.getTriggers().values().iterator();
-    while (it.hasNext()) {
-      Trigger<?> t = (Trigger<?>) it.next();
-      if (t.getClass().equals(GerritTrigger.class)) {
-        GerritTrigger gerritTrigger = (GerritTrigger) t;
-        return gerritTrigger;
-      }
-    }
-    return null;
+  private GerritTrigger getGerritTrigger(Run<?, ?> build) {
+    Job<?,?> project = build.getParent();
+    return (GerritTrigger) ParameterizedJobMixIn.getTrigger(project, GerritTrigger.class);
   }
 
   @Override
-  public boolean perform(AbstractBuild build, Launcher launcher,
-      BuildListener listener) throws IOException, InterruptedException {
+  public void perform(Run<?,?> build, FilePath path, Launcher launcher,
+      TaskListener listener) throws IOException, InterruptedException {
 
     Result result = build.getResult();
     // do not report results for aborted builds
     // TODO: possibly allow saving report even if build did not complete?
```

Comment:
Yes, a checkbox may be an option here. The main reason why I needed to touch this is usage from pipeline jobs, where you want to report results from various stages in a single build.